### ISSUE TITLE
VGM import doesn't determine start of data correctly

### ISSUE THREAD

**Issue by OPNA2608** [rerrahkr/YM2608-Tone-Editor#22]
The VGM data stream is currently hardcoded to be expected at `0x100`:
https://github.com/rerrahkr/YM2608-Tone-Editor/blob/0a076eaa853d2a61156af9c45d9ee8fedde6d268/src/io/vgm_io.cpp#L25-L27

This is a bad assumption. The [VGM specification](https://vgmrips.net/wiki/VGM_Specification) says:
![image](https://user-images.githubusercontent.com/23431373/217018102-467c4b95-73e9-49d4-be3e-881e35b00f5b.png)

So the correct flow for getting the start of the data should be (excluding read checks):
```cpp
uint32_t version = container.readUint32(0x08);
size_t csr = 0x40;

if (version >= 0x150)
    csr += container.readUint32(0x34);
```

Some examples of VGM files that don't get read correctly:

[YU-NO: Imperial City](https://vgmrips.net/packs/pack/kono-yo-no-hate-de-koi-wo-utau-shoujo-yu-no-nec-pc-9801#78-imperial-capital)
VGM version is `0x170`, VGM data offset is `0xAC`. Real start of VGM data is at `0xE0`.
A YM2608 port 0 write (command `0x56`) starts at `0xFE`, `0x100` has the last argument to that command
-> `0x70`, which is being interpreted as a 1-sample wait. All the parsing afterwards is likely very wrong.

[Briganty: Coda "BRIGANTY" (Ending)](https://vgmrips.net/packs/pack/briganty-the-roots-of-darkness-nec-pc-9801-opna#19-coda-briganty-ending)
VGM version is `0x170`, VGM data offset is `0xAC`. Real start of VGM data is at `0xE0`.
A YM2608 port 0 write (command `0x56`) starts at `0xFE`, `0x100` has the last argument to that command
-> `0x00`, which causes the VGM parser to abort due to an unknown command

**Comment by rerrahkr** [comment]
Thank you for your report. VGM Data Offset is the number of bytes from 0x34, so I will fix this as:

```diff
diff --git a/src/io/vgm_io.cpp b/src/io/vgm_io.cpp
index 0667a59..980adfd 100644
--- a/src/io/vgm_io.cpp
+++ b/src/io/vgm_io.cpp
@@ -11,7 +11,7 @@ VgmIo::VgmIo() : AbstractSongFileIo("vgm", "VGM file") {}
 
 std::vector<TonePtr> VgmIo::load(BinaryContainer& container) const
 {
-    if (container.size() < 256) throw FileCorruptionError(io::FileType::SongFile);
+	if (container.size() < 0x40) throw FileCorruptionError(io::FileType::SongFile);
 
     std::string ident = container.readString(0, 4);
     if (ident != "Vgm ") throw FileCorruptionError(io::FileType::SongFile);
@@ -20,9 +20,12 @@ std::vector<TonePtr> VgmIo::load(BinaryContainer& container) const
     uint32_t eof = container.readUint32(4);
     if (container.size() - 4 != eof) throw FileCorruptionError(io::FileType::SongFile);
 
+	uint32_t version = container.readUint32(0x08);
+	size_t csr = (version < 0x150) ? 0x40 : (0x34 + container.readUint32(0x34));
+	if (container.size() <= csr) throw FileCorruptionError(io::FileType::SongFile);
+
 	RegisterRecorder rec(44100);
 
-    size_t csr = 0x100;
 	for (bool flag = true; flag;) {
 		switch (uint8_t com = container.readUint8(csr++)) {
 		case 0x52:	// YM2612 Port A
```

**Comment by rerrahkr** [comment]
I fixed this in ee95aad.
